contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -53,83 +68,83 @@ contract ItemsRolesRegistryFacet is Modifiers, ISftRolesRegistry, ERC1155Holder
+            _grantRoleData.nonce,
+            UNIQUE_ROLE,
+            _grantRoleData.tokenAddress,
+            _grantRoleData.tokenId,
+            _grantRoleData.tokenAmount,
+            _grantRoleData.grantor,
+            _grantRoleData.grantee,
+            _grantRoleData.expirationDate,
+            _grantRoleData.revocable,
+            _grantRoleData.data
         );
     }
 
-    function _deposit(uint256 _nonce, DepositInfo memory _depositInfo) internal {
-        require(_depositInfo.tokenAmount > 0, "ItemsRolesRegistryFacet: tokenAmount must be greater than zero");
-        require(s.itemsDeposits[_nonce].grantor == address(0), "ItemsRolesRegistryFacet: deposit already exists");
-
-        s.itemsDeposits[_nonce] = _depositInfo;
-
-        emit Deposited(_nonce, _depositInfo.tokenAddress, _depositInfo.tokenId, _depositInfo.tokenAmount, _depositInfo.grantor);
-
-        _transferFrom(_depositInfo.grantor, address(this), _depositInfo.tokenAddress, _depositInfo.tokenId, _depositInfo.tokenAmount);
-    }
-
-    function revokeRoleFrom(uint256 _nonce, bytes32 _role) external override {
-        // revoke(depositId, role1)
-        RoleData memory _roleData = s.itemsRoleAssignments[_nonce];
-        require(_roleData.grantee != address(0), "ItemsRolesRegistryFacet: role does not exist");
-        DepositInfo memory _depositInfo = s.itemsDeposits[_nonce];
+    function revokeRoleFrom(uint256 _depositId, bytes32 _role, address _grantee) external override validRoleAndGrantee(_role, _grantee, _depositId) {
+        RoleData memory _roleData = s.itemsRoleAssignments[_depositId];
+        DepositInfo memory _depositInfo = s.itemsDeposits[_depositId];
 
         address caller = _findCaller(_roleData, _depositInfo);
         if (_roleData.expirationDate > block.timestamp && !_roleData.revocable) {
             // if role is not expired and is not revocable, only the grantee can revoke it
-            require(caller == _roleData.grantee, "ItemsRolesRegistryFacet: role is not revocable or caller is not the approved");
+            require(caller == _roleData.grantee, "ItemsRolesRegistryFacet: depositId is not expired or is not revocable");
         }
 
+        uint256 _gotchiId = s.depositIdToItemIdToGotchiId[_depositId][_depositInfo.tokenId];
 
-        if (s.itemIdToDelegationIdToGotchiId[_nonce][_depositInfo.tokenId] != 0) {
-            uint256 _gotchiId = s.itemIdToDelegationIdToGotchiId[_nonce][_depositInfo.tokenId];
-            LibItems.removeFromParent(address(this), _gotchiId, _depositInfo.tokenId, 1);
-            IEventHandlerFacet(s.wearableDiamond).emitTransferSingleEvent(msg.sender, address(this), msg.sender, _depositInfo.tokenId, 1);
-            emit LibERC1155.TransferFromParent(address(this), _gotchiId, _depositInfo.tokenId, 1);
-            delete s.itemIdToDelegationIdToGotchiId[_nonce][_depositInfo.tokenId];
-        }
+        if (_gotchiId != 0) {
+            _unequipWearable(_gotchiId, _depositInfo.tokenId);
```

Comment:
Don't we need to pass the `tokenAmount`? What if there are more than one NFT equipped (gloves)?

Reply:
In the actual logic we are only allowing to equip 1 deposit per gotchi. There would not be more than one gotchi for the same wearable balance delegated. It would require to revoke it and equip again